graph-error/src/error.rs

```diff
@@ -268,11 +271,15 @@ impl WithGraphError for reqwest::blocking::Response {
         let code = self.status();
         if code.is_client_error() || code.is_server_error() {
             let headers = Some(GraphHeaders::from(&self));
-            let error_message = self.json().unwrap_or_default();
+            let response_raw: serde_json::Value = self.json().unwrap_or_else(|err| {
+                serde_json::Value::String(format!("unable to parse response as JSON: {}", err))
+            });
+            let error_message = serde_json::from_value(response_raw.clone()).unwrap_or_default();
```

Comment:
I do have one question about this. If there is an error for Microsoft Graph inside that the response body and it is not in JSON format, will this actually give us the that error in a string format? This seems to me like it would just tell us the serde error which is probably something along the lines of `could not parse`.

I would think for this to work where we want to get the error, if there is one, that is for the Graph api then we would want to discard any parsing errors and just get it as a string.

So for instance, Microsoft Graph occasionally returns text inside response bodies when the response is a 204 No Content. And while this is strange, the way we handle this is just by wrapping the raw response in a string. Below is similar to how we would do it for 204 No Content responses:

```rust
response
    .text()
    .await
    .map(|s| serde_json::from_str(s.as_str()).unwrap_or(serde_json::Value::String(s)))
    .or_else(|err| serde_json::Value::String(format!("unable to get raw response: {}", err)));
```

If you still wanted to check if you could get JSON from it, you would just do the same thing you already are and see if serde will deserialize the string into a type.